src/ir.cpp

```diff
@@ -12388,26 +12425,25 @@ static IrInstruction *ir_analyze_array_mult(IrAnalyze *ira, IrInstructionBinOp *
 }
 
 static IrInstruction *ir_analyze_merge_error_sets(IrAnalyze *ira, IrInstructionBinOp *instruction) {
-    ZigType *op1_type = ir_resolve_type(ira, instruction->op1->child);
-    if (type_is_invalid(op1_type))
-        return ira->codegen->invalid_instruction;
-
-    if (op1_type->id != ZigTypeIdErrorSet) {
-        ir_add_error(ira, instruction->op1,
-                buf_sprintf("expected error set type, found '%s'", buf_ptr(&op1_type->name)));
+    ZigType *op1_type = ir_resolve_type(ira, instruction->op1->child, ZigTypeIdErrorSet);
+    if (type_is_invalid(op1_type)) {
+        if (ira->codegen->errors.length != 0) {
```

Comment:
Doesn't this make the assumption that if any errors have been reported, then `ir_resolve_type` reported it.
Is this always true? Couldn't some other code report an error, then an invalid type is passed to `ir_resolve_type` (which cause `ir_resolve_type` to not write an error).